src/components/App/ExpressRecorder.tsx

```diff
@@ -421,37 +419,87 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
         const isProcessing = () => processingCamera || processingScreen;
         this.setState({ processing: isProcessing() }, () => {
             if (cameraBlobs) {
+                // handle Safari compatibility
+                const browserName = this.getBrowserName();
+                const mimeType = browserName === "Safari" ? "video/mp4" : "video/webm";
+                const cameraBlob = new Blob(cameraBlobs, { type: mimeType });
                 // handle chrome blob duration issue
-                const cameraBlob = new Blob(cameraBlobs, { type: "video/webm" });
-                fixWebmDuration(cameraBlob, duration, (fixedBlob: Blob) => {
+                if (mimeType === "video/webm") {
+                    fixWebmDuration(cameraBlob, duration, (fixedBlob: Blob) => {
+                        processingCamera = false;
+                        this.setState({
+                            cameraBlob: fixedBlob,
+                            processing: isProcessing(),
+                            doPlayback: !isProcessing()
+                        });
+                        if (!isProcessing()) {
+                            this.dispatcher.dispatchEvent(RecorderEvents.recordingEnded);
+                        }
+                    });
+                } else {
+                    // Safari does not support fix-webm-duration
                     processingCamera = false;
                     this.setState({
-                        cameraBlob: fixedBlob,
+                        cameraBlob: cameraBlob,
                         processing: isProcessing(),
                         doPlayback: !isProcessing()
                     });
                     if (!isProcessing()) {
                         this.dispatcher.dispatchEvent(RecorderEvents.recordingEnded);
                     }
-                });
+                }
             }
             if (screenBlobs) {
-                const screenBlob = new Blob(screenBlobs, { type: "video/webm" });
-                fixWebmDuration(screenBlob, duration, (fixedBlob: Blob) => {
+                const browserName = this.getBrowserName();
+                const mimeType = browserName === "Safari" ? "video/mp4" : "video/webm";
+                const screenBlob = new Blob(screenBlobs, { type: mimeType });
```

Comment:
reuse this variables - lift them outside this if scope and use them above as well